### ISSUE TITLE
Unable to edit with heredoc

### ISSUE THREAD

**Issue by bryan-kramer** [emacs-php/php-mode#490]
I have inherited a file with several large heredoc blocks (SQL for use by PDO). 
It is proving impossible to edit PHP code - trying to type `->` or `(`  or enter causes php mode to fail without inserting the character. It makes editing impossible. 

The messages include 
```
Error running timer ‘show-paren-function’: (error "parse-sexp-propertize-function did not move syntax-propertize--done")
Error during redisplay: (internal--syntax-propertize 10201) signaled (error "Too much matching reentrancy")
Error during redisplay: (internal--syntax-propertize 10201) signaled (error "parse-sexp-propertize-function did not move syntax-propertize--done") [16 times]
Error during redisplay: (internal--syntax-propertize 9560) signaled (error "parse-sexp-propertize-function did not move syntax-propertize--done")
```

I am using commit a9ee3afd6aa4b25457ac553d44dec88782a38f11 but I have been seeing it before I pulled the latest.

Emacs version
```
GNU Emacs 27.0.50 (build 20, x86_64-pc-linux-gnu, GTK+ Version 3.22.26) of 2018-11-30
```

**Comment by imabug** [comment]
Also running into this with one of my PHP scripts with a heredoc block.

PHP Mode 1.20.0 of 2018-12-05
GNU Emacs 27.0.50 (build 3, x86_64-pc-linux-gnu, GTK+ Version 3.24.1) of 2019-01-02

**Comment by zonuexe** [comment]
@bryan-kramer @imabug 
Sorry for not commenting on this issue.  
This issue was resolved in #496 and released yesterday as [v1.21.2](https://github.com/emacs-php/php-mode/releases/tag/v1.21.2).

If this error recurs, please reopen it.  Thank you.
